tests/firebird/firebird_integration_test.go

```diff
@@ -0,0 +1,283 @@
+		if err != nil {
+			t.Fatalf("unable to prepare insert statement: %v", err)
+		}
+		defer stmt.Close()
+
+		numPlaceholders := strings.Count(insertStatement, "?")
+		if numPlaceholders == 0 {
+			t.Fatalf("insert statement has no placeholders '?' but params were provided")
+		}
+		for i := 0; i < len(params); i += numPlaceholders {
+			end := i + numPlaceholders
+			if end > len(params) {
+				end = len(params)
+			}
+			batchParams := params[i:end]
+
+			_, err = stmt.ExecContext(ctx, batchParams...)
+			if err != nil {
+				t.Fatalf("unable to insert test data row with params %v: %v", batchParams, err)
+			}
+		}
+	}
+
+	return func(t *testing.T) {
+		t.Helper()
+		isNotFoundError := func(err error) bool {
+			if err == nil {
+				return false
+			}
+			errMsg := strings.ToLower(err.Error())
+			return strings.Contains(errMsg, "does not exist") || strings.Contains(errMsg, "not found") || strings.Contains(errMsg, "is not defined")
+		}
+
+		_, err := db.ExecContext(ctx, fmt.Sprintf("DROP TRIGGER BI_%s_ID;", tableName))
+		if err != nil && !isNotFoundError(err) {
+			t.Logf("Could not drop trigger (this might be ok): %s", err)
+		}
+		_, err = db.ExecContext(ctx, fmt.Sprintf("DROP TABLE %s;", tableName))
+		if err != nil && !isNotFoundError(err) {
+			t.Errorf("Teardown failed to drop table: %s", err)
+		}
+		_, err = db.ExecContext(ctx, fmt.Sprintf("DROP GENERATOR GEN_%s_ID;", tableName))
+		if err != nil && !isNotFoundError(err) {
+			t.Logf("Could not drop generator (this might be ok): %s", err)
+		}
+	}
+}
+
+func getFirebirdParamToolInfo(tableName string) ([]string, string, string, string, string, string, []any) {
+	createStatements := []string{
+		fmt.Sprintf("CREATE TABLE %s (ID INTEGER NOT NULL PRIMARY KEY, NAME VARCHAR(255));", tableName),
+		fmt.Sprintf("CREATE GENERATOR GEN_%s_ID;", tableName),
+		fmt.Sprintf(`
+			CREATE TRIGGER BI_%s_ID FOR %s
+			ACTIVE BEFORE INSERT POSITION 0
+			AS
+			BEGIN
+				IF (NEW.ID IS NULL) THEN
+					NEW.ID = GEN_ID(GEN_%s_ID, 1);
+			END;
+		`, tableName, tableName, tableName),
+	}
+
+	insertStatement := fmt.Sprintf("INSERT INTO %s (NAME) VALUES (?);", tableName)
+	toolStatement := fmt.Sprintf("SELECT ID, NAME FROM %s WHERE ID = ? OR NAME = ?;", tableName)
+	idParamStatement := fmt.Sprintf("SELECT ID, NAME FROM %s WHERE ID = ?;", tableName)
+	nameParamStatement := fmt.Sprintf("SELECT ID, NAME FROM %s WHERE NAME IS NOT DISTINCT FROM ?;", tableName)
+	arrayToolStatement := fmt.Sprintf("SELECT ID, NAME FROM %s WHERE ID IN (?) ORDER BY ID;", tableName)
+
+	params := []any{"Alice", "Jane", "Sid", nil}
+	return createStatements, insertStatement, toolStatement, idParamStatement, nameParamStatement, arrayToolStatement, params
+}
+
+func getFirebirdAuthToolInfo(tableName string) ([]string, string, string, []any) {
+	createStatements := []string{
+		fmt.Sprintf("CREATE TABLE %s (ID INTEGER NOT NULL PRIMARY KEY, NAME VARCHAR(255), EMAIL VARCHAR(255));", tableName),
```

Comment:
```suggestion
		fmt.Sprintf("CREATE TABLE %s (id INTEGER NOT NULL PRIMARY KEY, name VARCHAR(255), email VARCHAR(255));", tableName),
```

Same with this, please update accordingly :)